contentcuration/contentcuration/models.py

```diff
@@ -1720,6 +1720,78 @@ class File(models.Model):
 
     objects = CustomManager()
 
+    _edit_filter = Q()
+    for tree_name in CHANNEL_TREES:
+        _edit_filter |= Q(
+            **{
+                "editable_channels__{}__tree_id".format(tree_name): OuterRef(
+                    "contentnode__tree_id"
+                )
+            }
+        )
+        _edit_filter |= Q(
+            **{
+                "editable_channels__{}__tree_id".format(tree_name): OuterRef(
+                    "assessment_item__contentnode__tree_id"
+                )
+            }
+        )
+
+    _view_filter = Q()
+    for tree_name in CHANNEL_TREES:
+        _view_filter |= Q(
+            **{
+                "view_only_channels__{}__tree_id".format(tree_name): OuterRef(
+                    "contentnode__tree_id"
+                )
+            }
+        )
+        _view_filter |= Q(
+            **{
+                "view_only_channels__{}__tree_id".format(tree_name): OuterRef(
+                    "assessment_item__contentnode__tree_id"
+                )
+            }
+        )
+
+    @classmethod
+    def filter_edit_queryset(cls, queryset, user):
+        user_id = not user.is_anonymous() and user.id
+        user_queryset = User.objects.filter(id=user_id)
```

Comment:
Since we're not using anything from the actual User model, perhaps we can just go straight to the querysets for the related model tables, `view_only_channels` and `editable_channels`, and save one join on the DB side?